episodes/chapter8.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Work on your own project"
+teaching: 10
+exercises: 1
+---
+
+:::::::::::::::::::::::::::::::::::::: questions
+
+- How can you use vantage6 in your own project?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Start working on your own research project
+
+OR
+
+- Complete some advanced challenges
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+# Working on your own project
+
+In this chapter, we will give you the opportunity to work on your own project. Feel free
+to ask questions and discuss your project with the instructors. We are here to help you!
+
+If you prefer, you can also complete some advanced challenges that we have prepared for
+you.
+
+# Advanced challenges
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge: Add privacy filters to your algorithm
+
+In the [previous lesson](./algorithm_development.md), you have created a simple
+algorithm. Now, you can add privacy filters to your algorithm.
+
+- Create an algorithm that does not return the results unless there are more than 10
+  data points.
+- Make this value configurable by the node administrator. Hint: look up the node
+  configuration options in the [documentation](https://docs.vantage6.ai) to provide
+  environment variables to your algorithm.
+- What else could you do to protect the privacy of the data?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge: Document your algorithm
+
+In the [previous lesson](./algorithm_development.md), you have created a simple
+algorithm. For this challenge, learn how to document your algorithm.
+
+- Make sure a documentation template is available in the algorithm repository. If it is
+  not, you can generate it using the `v6 algorithm` command.
+- Try out how you can run the documentation locally in your browser, and adapt it to
+  your needs.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge: Visualize the results of your algorithm in the UI
+
+In the [previous lesson](./algorithm_development.md), you have created a simple
+algorithm. In this challenge, modify the algorithm in the algorithm store to include
+a visualization of the results.
```

Comment:
Do they do this using the UI? Also we might want to provide some hints here. e.g.: the output always needs to a pandas dataframe json format. Also the documentation on the `algorithm_store.json` is missing in the docs? I guess this section needs to be expanded?: https://docs.vantage6.ai/en/main/algorithms/develop.html#calling-your-algorithm-from-vantage6

Reply:
- Yes I added that they should do this in the UI
- The output doesn't need to be dataframe json necessarily - a flat dictionary will be visualized as a key-value table.
- Yes that section definitely needs to be expanded, that's already on my list. So there it will mention that the store json is missing.